apps/wallet/src/ui/app/staking/validators/SelectValidatorCard.tsx

```diff
@@ -34,15 +54,28 @@ export function SelectValidatorCard() {
         setSelectedValidator((state) => (state !== address ? address : null));
     };
 
+    const handleSortByKey = (key: 'name' | 'apy') => {
+        setSortKey(key);
+        setSortAscending(!sortAscending);
+    };
+
     const validatorList = useMemo(() => {
         if (!validatorsData) return [];
-        return validatorsData.validators.fields.active_validators.sort((a, b) =>
-            getName(a.fields.metadata.fields.name) >
-            getName(b.fields.metadata.fields.name)
-                ? 1
-                : -1
-        );
-    }, [validatorsData]);
+
+        return validatorsList(
+            validatorsData.validators.fields.active_validators,
+            +validatorsData.epoch
+        ).sort((a, b) => {
+            if (sortKey === 'name') {
+                return sortAscending
+                    ? collator.compare(a[sortKey], b[sortKey])
```

Comment:
Probably easier to use `String.prototype.localeCompare`:
`a[sortKey].localeCompare(b[sortKey])`